crates/telio-relay/src/derp/mod.rs

```diff
@@ -565,11 +565,18 @@ impl Runtime for State {
 
         match &mut self.conn {
             Some(c) => {
```

Comment:
Can we have Some(self.conn) && None(self.channel)?

Also, on the Some(conn) case there's no connection attempt, it is done only on the None(conn) case where you added the check ([line 616)](https://github.com/NordSecurity/libtelio/pull/91/files#diff-51a48b71eedfdd3893868ac7fb36eb992316673851e6a379ee230b84058a1843R616), isn't that enough?

if self.channel.is_none() {
                    return (update.await)(self).await;
                }

Reply:
>Can we have Some(self.conn) && None(self.channel)?

I don't think so, but we need a channel here, not an option, so I need to get it somehow. As I said, if you don't like this solution, I can move it to the large `match` statement, or replace with a flag, or refactor it in some other way